### PR TITLE
check mono, z3, and 64 bit OS

### PR DESCRIPTION
**Pull request by goldylucks**
- validate above prerequisites
- show relevant error message to user if needed
- add platform utils
- refactor app updater with platform utils

QA
===

I don't know how to QA Z3 not working. maybe @somdoron have an idea.

#### LINUX
- [ ] run wallet in 64 bit machine - no errors
- [ ] run wallet in 32 bit machine - see "Zen-node can only run on 64 bit OS

#### WINDOWS
- [ ] run wallet in 64 bit machine - no errors
- [ ] run wallet in 32 bit machine - see "Zen-node can only run on 64 bit OS

#### OSX
- [ ] run wallet with latest mono - no errors
- [ ] uninstall mono and run wallet - see "please install mono"
- [ ] downgrade mono to major version 4.x.x and run wallet - see "Old version of mono, please upgrade" (minor and patch versions doesn't matter)
- [ ] downgrade mono to major version 5.9.x and run wallet - see "Old version of mono, please upgrade"
(patch version doesn't matter)
- [ ]  run wallet in 64 bit machine - no errors
- [ ]  run wallet in 32 bit machine - see "Zen-node can only run on 64 bit OS

### CODE REVIEW

File: app/utils/prereqCheck.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    process.exit(1)
+  }
+}
+
+function getZ3PathForPlatform() {
+  const pathToDir = path.join(getZenNodePath(), 'Release')
+  let filename
+  if (isWindows) { filename = 'z3.exe' }
+  if (isOsx) { filename = 'z3-osx' }
+  if (isLinux) { filename = 'z3-linux' }
```
Comment (somdoron) [inline app/utils/prereqCheck.js]: this is function and you using it like variable, they are all true, so it is always will be z3-linux

File: app/utils/prereqCheck.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+function validateMono() {
+  if (!isOsx()) {
+    console.log('mono version check passed (not needed for non OSX)')
+    return
+  }
+  const { status, msg } = getMonoStatus()
+  if (status === ERR_STATUS) {
+    console.error(msg)
+    dialog.showErrorBox(
+      'Mono check failed',
```
Comment (Cremba) [inline app/utils/prereqCheck.js]: this is never printed instead there is an error ENOENT, which usually mean that the mono version is not updated or that mono is not installed

Comment (goldylucks) [reply app/utils/prereqCheck.js]: Thank you for reporting @Cremba , it works for me on ubuntu when I trigger the error deliberately by passing a wrong path.

Will look into it.

Comment (goldylucks) [reply app/utils/prereqCheck.js]: [PR](https://github.com/zenprotocol/zenwallet/pull/82) attempted fix

**Pull request merged**